cpp/src/umap/has_nan.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <cuda_runtime.h>
```

Comment:
Let's put this file in a more reusable place. Either in the utils in the algorithms layer or the prims layer.

Reply:
Yes, that was temporary